gax/retry_loop_test.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gax/retry_loop.h"
+#include "google/longrunning/operations.pb.h"
+#include "googletest/include/gtest/gtest.h"
+#include "gax/backoff_policy.h"
+#include "gax/call_context.h"
+#include "gax/retry_policy.h"
+#include <chrono>
+
+namespace {
+using namespace ::google;
+
+std::unique_ptr<gax::RetryPolicy> ErrCountRetryFactory(int n) {
+  return std::unique_ptr<gax::LimitedErrorCountRetryPolicy>(
+      new gax::LimitedErrorCountRetryPolicy(n));
+}
+
+TEST(RetryLoop, Basic) {
+  // Hack to count the number of backoff attempts.
+  static int delay_count = 0;
+
+  class DummyBackoffPolicy : public gax::BackoffPolicy {
+   public:
+    std::chrono::microseconds OnCompletion() override {
+      delay_count++;
+      return std::chrono::microseconds(0);
+    }
+    std::unique_ptr<gax::BackoffPolicy> clone() const override {
+      return std::unique_ptr<DummyBackoffPolicy>(new DummyBackoffPolicy());
```

Comment:
If this is not used, please provide a dummy implementation (like `return NULL;`) instead and maybe an inline comment like `//not used`

Reply:
I'm a little uncomfortable with that. In an ideal world, `clone` would be inferrable or auto-generated or somehow defined once in a way that just returns a new unique_ptr and invokes the copy constructor. Returning `nullptr` breaks that default behavior.